src/app/clusters/bindings/bindings.cpp

```diff
@@ -52,7 +52,7 @@ class BindingTableAccess : public AttributeAccessInterface
     CHIP_ERROR ReadBindingTable(EndpointId endpoint, AttributeValueEncoder & encoder);
     CHIP_ERROR WriteBindingTable(const ConcreteDataAttributePath & path, AttributeValueDecoder & decoder);
 
-    CHIP_ERROR NotifyBindingsChanged();
+    CHIP_ERROR NotifyBindingsChanged(FabricIndex accessingFabricIndex = 0);
```

Comment:
is there a need to have a default? seems like this might allow a caller to forget to set it - in which case some sort of "unknown fabric index" value would be better. I suggest not having a default unless you absolutely need it

Reply:
Somewhat recently this PR: https://github.com/sharadb-amazon/connectedhomeip/commit/3c0c8753a4d660507f822ec5e28dae3a1e4b8875#diff-cff38b6dd18f2dee82a0430ab4d8d1199a330559936fb17e1be1bc8dc4131c54 added a OnListWriteEnd function. I think we should be ok passing in a 0 for a fabricIndex when that function calls NotifyBindingsChanged, which is why I had added the default value. If passing 0 makes sense (let me know if it doesn't), I have changed to code, so there are no defaults now, but we explicitly pass 0 in the implementation of OnListWriteEnd.

Reply:
OnListWriteEnd was added to avoid that the previous implementation sent out an event when the list was cleared and one when the "new" binding list was written again. 

The notify function call inside `if (!path.IsListOperation())` should not happen and it is unclear if this situation is even possible: https://github.com/project-chip/connectedhomeip/issues/24377

My assumption for this to work we would want to return the correct accessingFabricIndex in OnListWriteEnd and not just 0?

Reply:
@ReneJosefsen - Thanks, you're right, I'll need to return a valid accessingFabricIndex in OnListWriteEnd. 

If you are familiar with this version of the BindingTableAccess class and how the state of its objects are handled, can you tell me if that can be done by capturing the accessingFabricIndex in WriteBindingTable as a new data member (say mAccessingFabricIndex) of the BindingTableAccess object, and then passing the value of that data member in OnListWriteEnd?

Reply:
Diff updated.